skymap_scanner/utils/data_handling.py

```diff
@@ -53,10 +54,24 @@ def stage_file(self, basename: str):
         local_destination_path = self.staging_path / basename
         http_source_path = f"{self.remote_path}/{basename}"
         # not sure why we use the -O pattern here
-        cmd = f"wget -nv -t 5 -O {local_destination_path} {http_source_path}"
-        return_value = os.system(cmd)
-        if return_value != 0 or not local_destination_path.is_file():
-            raise RuntimeError(f"Failed to retrieve data from remote source:\n-> {cmd}")
+        cmd = [
+            "wget",
+            "-nv",
+            "-t",
+            "5",
+            "-O",
+            str(local_destination_path),
+            http_source_path,
+        ]
+        try:
+            subprocess.run(cmd, check=True)
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError(f"Subprocess `wget` exited with status:\n -> {e}")
+
+        if not local_destination_path.is_file():
+            raise RuntimeError(
+                f"Subprocess `wget` succeeded but the resulting file is invalid:\n-> {cmd}"
+            )
 
     def get_filename(self, basename: str) -> str:
         """Look up basename under the local paths and the staging path and returns the first valid filename.
```

Comment:
Does `basename` refer to the filename, similar to bash's `$(basename $FILE)`?

Reply:
Yes. I call `basename` something like `bar.ext` and `filename` something like `/foo/bar.ext`.

Honestly I find "basename" not very intuitive. I would rather call `bar.ext` a `filename` and `/foo/bar.ext` a `filepath`, but then confusion may arise between paths in the form of `str` and paths in the form of `Path` objects... so I don't have a naming convention I am fully satisfied with right now.

Reply:
> I would rather call bar.ext a filename and /foo/bar.ext a filepath

This is the convention I usually follow. If the variable name has "name" it's a string. If it has "path" it's either a string or a `Path`, but if it is a str, then it can be cast to a `Path` instance